chisel/src/executor.rs

```diff
@@ -357,6 +384,44 @@ fn format_token(token: Token) -> String {
     }
 }
 
+/// Formats a [pt::EventDefinition] into an inspection message
+///
+/// ### Takes
+///
+/// An borrowed [pt::EventDefinition]
+///
+/// ### Returns
+///
+/// A formatted [pt::EventDefinition] for use in inspection output.
+///
+/// TODO: Verbosity option
+fn format_event_definition(event_definition: &pt::EventDefinition) -> Result<String> {
+    let event_name = event_definition.name.as_ref().expect("Event has a name").to_string();
+    let inputs = event_definition
+        .fields
+        .iter()
+        .map(|param| {
+            let name = param
+                .name
+                .as_ref()
+                .map(ToString::to_string)
+                .unwrap_or_else(|| "<anonymous>".to_string());
+            let kind = Type::from_expression(&param.ty)
+                .and_then(Type::into_builtin)
+                .ok_or_else(|| eyre::eyre!("Invalid type in event {event_name}"))?;
+            Ok(ethabi::EventParam { name, kind, indexed: param.indexed })
+        })
+        .collect::<Result<Vec<_>>>()?;
+    let event = ethabi::Event { name: event_name, inputs, anonymous: event_definition.anonymous };
+
+    Ok(format!(
+        "Type: {}\n├ Name: {}\n└ Signature: {:?}",
+        Paint::red("event"),
+        Paint::cyan(&event.name),
```

Comment:
Would also be sweet if we could output the event's parameters here.

Reply:
Absolutely agree! It's a bit out-of-scope for me since I'm not really familiar with the codebase and the crates it uses yet, but maybe in a follow-up PR? :)

Reply:
No worries, okay with me if you want to add it in a separate PR. If you want to add it in this PR, you already have all of the info you need in the `event` definition: 
```suggestion
        SolidityHelper::highlight(&format!(
            "{}({})",
            &event.name,
            &event
                .inputs
                .iter()
                .map(|param| format!(
                    "{}{}{}",
                    param.kind,
                    if param.indexed { " indexed" } else { "" },
                    if param.name.is_empty() {
                        String::default()
                    } else {
                        format!(" {}", &param.name)
                    },
                ))
                .collect::<Vec<_>>()
                .join(", ")
        ))
```

Reply:
![Screenshot from 2023-03-16 12-00-36](https://user-images.githubusercontent.com/8406232/225680358-e4ec5ea4-add4-47d9-a3f3-2ccddffe59c3.png)

Reply:
Thanks for writing the code for this! Will then just commit your change.